.github/workflows/build-image.yml

```diff
@@ -121,19 +123,47 @@ jobs:
     if: ${{ needs.detect-changes.outputs.cuda == 'true' }}
     strategy:
       fail-fast: false
+      # multi-OS build matrix: creates 4 images (2 platforms × 2 OS variants)
+      # - RHEL (ubi9): default/production image (no suffix in final tags)
+      # - Ubuntu (24.04): alternative base for compatibility testing (-ubuntu suffix)
+      # resulting manifests: cuda-dev (rhel) and cuda-dev-ubuntu
       matrix:
         include:
           - platform: linux/amd64
-            arch: amd64
-    runs-on: vllm-runner
+            runner: vllm-runner
+            os: rhel
+            base_image_suffix: ubi9
+          - platform: linux/arm64
+            runner: vllm-runner-arm
+            os: rhel
+            base_image_suffix: ubi9
+          - platform: linux/amd64
+            runner: vllm-runner
+            os: ubuntu
+            base_image_suffix: ubuntu24.04
+          - platform: linux/arm64
+            runner: vllm-runner-arm
+            os: ubuntu
+            base_image_suffix: ubuntu24.04
+    runs-on: ${{ matrix.runner }}
     steps:
+      - name: Free disk space (ARM only)
+        if: matrix.platform == 'linux/arm64'
+        uses: jlumbroso/free-disk-space@main
+        with:
+          tool-cache: false
+          android: true
+          dotnet: true
+          haskell: true
+          large-packages: true
+          docker-images: false
+          swap-storage: true
```

Comment:
Do we need this? Now we are using a managed runner for ARM builds we shouldn't need to free space and this will slow build times down